### PR TITLE
Add cost function

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR adds the function `get_costs()` to calculate pandemic costs from a `<daedalus_output>` object.

The best place to see the use case is in the newly added vignette, "pandemic_costs.Rmd".

Costs are calculated as:

## Economic costs

1. Closures: Daily GVA per sector $\times$ (1 - openess) $\times$ closure duration in days;
2. Absences: Daily GVA per sector, accounting for closures $\times$ proportion workers available, for each day; workers are available if not infectious, hospitalised, or dead.

## Education costs

Education costs are calculated as the GVA loss from closures and absences in the education sector. They are not included in the economic costs when presented to the user.

## Life years lost

Life expectancy for each age group $\times$ value of a statistical life. There doesn't seem to be any discounting for age.

## Other changes:

1. Added raw data on life expectancy and value of a life year (VLY) in multiple countries;
2. Converted (1) into package data `life_value` for age-group-wise data on the value of a statistical life in each country (VSL).
    - Number of countries reduced to 67 with Hong Kong and Taiwan removed due to lack of life expectancy or value data.
3. `<daedalus_country>` class automatically pulls in VSL data when created;
4. `<daedalus_output>` now includes total simulation time and openness coefficient data (used in cost calculation);
5. Pkg imports {data.table} for cost calculations.

### CODE REVIEW

**Comment by pratikunterwegs** [comment]
Weird case of the test workflow failing when the R CMD check workflow passes; the check workflow includes running the tests. Will look into it.

**Comment by pratikunterwegs** [comment]
Unsure what the issue is here as I can't reproduce this error locally. My next guess is that there's an issue with the workflow in either {spelling} or {testthat}.

**Comment by pratikunterwegs** [comment]
Removed {data.table} which fixes the failing coverage workflow.

**Review by richfitz — APPROVED** [review]
A couple of small suggestions to clarify the calculation, but I don't at all feel strongly on this

File: R/costs.R [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  ]
+
+  # set factor levels to keep order
+  total_deaths$age_group <- factor(
+    total_deaths$age_group,
+    levels = unique(total_deaths$age_group)
+  )
+  total_deaths <- stats::aggregate(
+    value ~ age_group, total_deaths, sum
+  )$value
```
[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply R/costs.R:111]: Will switch to `tapply()` here, lower-level is better.

File: R/costs.R [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  gva_loss[seq(closure_start, closure_end)] <- lapply(
+    gva_loss[seq(closure_start, closure_end)],
+    `*`, openness
+  )
+  gva_loss <- Reduce(`+`, gva_loss)
+
+  education_cost_absences <- gva_loss[i_EDUCATION_SECTOR]
+  economic_cost_absences <- sum(
+    gva_loss[-i_EDUCATION_SECTOR]
+  )
```
[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply R/costs.R:96]: I haven't full got whether this suggested snippet is accounting for the reduction in potential GVA due to closures, but will look at making it easier to read.

[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply R/costs.R]: I've updated this bit in https://github.com/jameel-institute/daedalus/pull/25/commits/4b3fa792180648ca9b81007af2eba162cc81a8b2 to use matrix multiplication, with a sub-matrix scaled by openness - hopefully that's easier to read.

File: R/class_country.R [hunk 3]
```diff
@@ -282,6 +288,12 @@ validate_daedalus_country <- function(x) {
         any.missing = FALSE, nrows = N_ECON_SECTORS, ncols = N_ECON_SECTORS
       ) && checkmate::test_subset(
         x$contacts_between_sectors, 0
+      ),
+    "Country `vsl` must be a 4-element vector of positive values" =
+      checkmate::test_numeric(
+        x$vsl,
+        len = N_AGE_GROUPS, any.missing = FALSE,
+        lower = 0, upper = 1e7 # reasonable upper limit
```
Comment (richfitz) [inline R/class_country.R:296]: Below on l406:

```
prepare_parameters.daedalus_country <- function(x) {
```

becomes

```
prepare_parameters.daedalus_country <- function(x, ...) {
```

to fix that build. New check in r-devel perhaps?

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/class_country.R (hunk 3 above)
Comment (pratikunterwegs) [reply R/class_country.R:296]: Yep just spotted this when the build failed!

**Pull request merged**